pkg/mcs/resource_manager/server/resource_group.go

```diff
@@ -164,10 +170,17 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
 }
 
 // RequestRU requests the RU of the resource group.
-func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64) *rmpb.GrantedRUTokenBucket {
+func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64, storage endpoint.ResourceGroupStorage) *rmpb.GrantedRUTokenBucket {
 	rg.Lock()
 	defer rg.Unlock()
-	if rg.RUSettings == nil {
+	grant := rg.requestRU(now, neededTokens, targetPeriodMs)
```

Comment:
How much overhead will it pay here? And how often is it called and do we need to call it so often?